### PR TITLE
RATIS-1901. Update Counter example to benchmark performance.

### PR DESCRIPTION
**Pull request by szetszwo**
See RATIS-1901

### CODE REVIEW

**Review by SzyWilliam — APPROVED** [review]
+1 the changes look good to me. Thanks @szetszwo for the benchmark!
I think we need to notice users the potential problems if we adopt a larger segment size.

File: ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java [hunk 1]
```diff
@@ -381,7 +381,7 @@ static void setPurgePreservationLogNum(RaftProperties properties, long purgePres
     }
 
     String SEGMENT_SIZE_MAX_KEY = PREFIX + ".segment.size.max";
-    SizeInBytes SEGMENT_SIZE_MAX_DEFAULT = SizeInBytes.valueOf("8MB");
+    SizeInBytes SEGMENT_SIZE_MAX_DEFAULT = SizeInBytes.valueOf("128MB");
```
[the remark this answers is not in the document]
Comment (szetszwo) [reply ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java:384]: How about changing it to a small size, say 32MB?  Currently, 8MB is very small.  The servers will roll the log frequently and create a lot of small files.

**Comment by szetszwo** [comment]
Hit the NPE fixed by RATIS-1920 in my test.  Will sync with the master branch.

File: ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (SzyWilliam) [reply ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java]: +1 8MB is small. Actually I think 128MB is good. Kafka even uses 1GB as the segment size. It's only that I did experiments earlier and found 32MB is the optimal for IOTDB default workload. The experiments show:
* (end to end throughput) improves substantially as segmented size grow from 8MB to 32MB, slowly from 32MB to 128MB.
* The memory usage in JVM young increases as the segmented size grow, even if we restricted the max-cached-size and max-cached-segments. It turned out to be the open segment.

[the remark this answers is not in the document]
Comment (szetszwo) [reply ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java]: It is good to know.  Let's change to 32MB then.  Filed RATIS-1924.

**Comment by szetszwo** [comment]
@SzyWilliam , thanks a lot for reviewing this!

**Pull request merged**